src/plugins/console/public/services/autocomplete.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { createGetterSetter } from '@kbn/kibana-utils-plugin/public';
+import type { HttpSetup } from '@kbn/core/public';
+import type {
+  ClusterGetComponentTemplateResponse,
+  IndicesGetAliasResponse,
+  IndicesGetDataStreamResponse,
+  IndicesGetIndexTemplateResponse,
+  IndicesGetMappingResponse,
+  IndicesGetTemplateResponse,
+} from '@elastic/elasticsearch/lib/api/types';
+import { API_BASE_PATH } from '../../common/constants';
+import {
+  Alias,
+  DataStream,
+  Mapping,
+  LegacyTemplate,
+  IndexTemplate,
+  ComponentTemplate,
+} from '../lib/mappings';
+import { DevToolsSettings, Settings } from './settings';
+
+interface MappingsApiResponse {
+  mappings: IndicesGetMappingResponse;
+  aliases: IndicesGetAliasResponse;
+  dataStreams: IndicesGetDataStreamResponse;
+  templates: {
+    legacyTemplates: IndicesGetTemplateResponse;
+    indexTemplates: IndicesGetIndexTemplateResponse;
+    componentTemplates: ClusterGetComponentTemplateResponse;
+  };
+}
+
+export class AutocompleteInfo {
+  public readonly alias: Alias;
```

Comment:
I see all following properties
```
  public readonly alias: Alias;
  public readonly mapping: Mapping;
  public readonly dataStream: DataStream;
  public readonly legacyTemplate: LegacyTemplate;
  public readonly indexTemplate: IndexTemplate;
  public readonly componentTemplate: ComponentTemplate;
```
implement 'get', 'load', 'clean' methods. Can we create a base generic class for that.